cachi2/core/package_managers/gomod.py

```diff
@@ -258,6 +264,23 @@ def release(self) -> str:  # type: ignore
         """Release name of the Go Toolchain, e.g. go1.20 ."""
         pass
 
+    @staticmethod
+    def _locate_toolchain(release: str) -> Optional[str]:
+        """Given a release locate an alternative Go toolchain.
+
+        Locate an alternative Go toolchain under the one of the following locations:
+            - /usr/local/go/                    for container environments (pre-installed)
+            - $XDG_CACHE_HOME/cachi2/go         for local environments (download & cache)
```

Comment:
Is `/usr/local/go/` purely our custom in-container location? Or is it the default system-wide Go location?

If it's just ours, maybe the container could pre-install to the ~/.cache location instead to simplify this

Reply:
> Is /usr/local/go/ purely our custom in-container location? Or is it the default system-wide Go location?

The path used by package managers on most distros will default to `/usr/lib/go`. I used `/usr/local` because the purpose of that path are host-local installations which is exactly what we're doing. We must not write to the default packaging system location.

> If it's just ours, maybe the container could pre-install to the $XDG_CACHE_HOME location instead to simplify this

We can't because:

1. `$XDG_CACHE_HOME` is usually not set, well, at least not on server-like distro spins
2. `~/.cache` points to the current container user's home, which in case of our build is root, but will most likely be something else during container runtime

and so I resorted to a widely known system location instead.

Reply:
You could set `USER` and `ENV HOME` or `ENV XDG_CACHE_HOME` in the Dockerfile, but the system location is probably easier

Reply:
Hmm, but doing that would mean that konflux build pipeline needs be adjusted accordingly so that one of those is explicitly set, wouldn't it? Unless we need that sort of hardening right now I'd like to keep it simple until we need to write that cache inside a container which I don't expect to be needed (hopefully ever).

Reply:
You can set these things in the Dockerfile, I think Tekton will respect them.

A separate problem could be that the git-clone task explicitly runs as root, so if prefetch runs as non-root it might lose the ability to write to the cloned source dir. So let's keep it as it is for now :smile: